MemoryInformation class depends on unavailable wmic command

**Describe the bug**  
The Windows wmic command has been deprecated for a while and has been removed from Windows 10 21H1+, Windows 11, Server 2022+.

The MemoryInformation class, in com.ibm.ws.kernel.service project, still uses it.

When the command is invoked, it produces this exception:

```
suite:java.lang.ExceptionInInitializerError
    at java.base/java.lang.Class.forName0(Native Method)
    at java.base/java.lang.Class.forName(Class.java:375)
Caused by: java.lang.RuntimeException: com.ibm.ws.kernel.service.util.MemoryInformationException: com.ibm.ws.kernel.service.util.OperatingSystemException: java.io.IOException: Cannot run program "wmic": CreateProcess error=2, The system cannot find the file specified
    at com.ibm.ws.health.manager.fat.FATSuite.<clinit>(FATSuite.java:155)
Caused by: com.ibm.ws.kernel.service.util.MemoryInformationException: com.ibm.ws.kernel.service.util.OperatingSystemException: java.io.IOException: Cannot run program "wmic": CreateProcess error=2, The system cannot find the file specified
    at com.ibm.ws.kernel.service.util.MemoryInformation.getAvailableMemoryWindows(MemoryInformation.java:696)
    at com.ibm.ws.kernel.service.util.MemoryInformation.getAvailableMemory(MemoryInformation.java:231)
    at com.ibm.ws.health.manager.fat.FATSuite.<clinit>(FATSuite.java:126)
Caused by: com.ibm.ws.kernel.service.util.OperatingSystemException: java.io.IOException: Cannot run program "wmic": CreateProcess error=2, The system cannot find the file specified
    at com.ibm.ws.kernel.service.util.OperatingSystem.executeProgramWithInput(OperatingSystem.java:182)
    at com.ibm.ws.kernel.service.util.OperatingSystem.executeProgram(OperatingSystem.java:129)
    at com.ibm.ws.kernel.service.util.MemoryInformation.getAvailableMemoryWindows(MemoryInformation.java:677)
Caused by: java.io.IOException: Cannot run program "wmic": CreateProcess error=2, The system cannot find the file specified
    at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
    at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
    at com.ibm.ws.kernel.service.util.OperatingSystem.executeProgramWithInput(OperatingSystem.java:151)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
    at java.base/java.lang.ProcessImpl.create(Native Method)
    at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:505)
    at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:158)
    at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)
```

**Steps to Reproduce**  

The problem was exposed by the following FATs:
- com.ibm.ws.health.manager_fat  
- com.ibm.ws.health.analyzer_fat

**Expected behavior**  
MemoryInformation class methods getTotalMemoryWindows() and getAvailableMemoryWindows() should return the appropriate memory sizes with throwing an exception.

**Diagnostic information:**  
 - OpenLiberty Version: [ 0.0.0.0 - 26.0.0.2 ]
 - Affected feature(s) [all versions of mpHealth ]
 - Java Version: [N/A]
 - server.xml configuration

--- a/dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/MemoryInformation.java
+++ b/dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/MemoryInformation.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2018, 2021 IBM Corporation and others.
+ * Copyright (c) 2018, 2026 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
@@ -650,51 +650,48 @@ public class MemoryInformation {
         return Long.parseLong(line);
     }
 
-    @FFDCIgnore({ OperatingSystemException.class })
     private long getTotalMemoryWindows() throws MemoryInformationException {
-        try {
-            List<String> lines = OperatingSystem.executeProgram("wmic", "os", "get", "totalvisiblememorysize", "/format:list");
-            for (String line : lines) {
-                // https://msdn.microsoft.com/en-us/library/aa394239(v=vs.85).aspx
-                if (line.startsWith("TotalVisibleMemorySize=")) {
-                    return processWmicLine(line) * 1024;
-                }
-            }
-            throw new MemoryInformationException(Tr.formatMessage(tc, "memory.information.unexpected", "wmic", lines));
-        } catch (OperatingSystemException e) {
-            throw new MemoryInformationException(e);
-        }
+        // This was previously implemented using wmic, which is deprecated 
+        // and removed in modern Windows (Windows 10 21H1+, Windows 11, Server 2022+)
+        // Use Java MXBean instead which is available in Java 8+.
+        return getTotalMemoryJDK();
     }
 
-    private long processWmicLine(String line) {
-        return Long.parseLong(line.substring(line.indexOf('=') + 1));
-    }
-
-    @FFDCIgnore({ OperatingSystemException.class })
+    @FFDCIgnore({ MemoryInformationException.class })
     private long getAvailableMemoryWindows() throws MemoryInformationException {
+        // wmic is deprecated and removed in modern Windows (Windows 10 21H1+, Windows 11, Server 2022+)
+        // Try PowerShell first to maintain the same level of detail (AvailableBytes + CacheBytes)
         try {
-            long available = 0;
-            List<String> lines = OperatingSystem.executeProgram("wmic", "path", "Win32_PerfFormattedData_PerfOS_Memory", "get",
-                                                                "/format:list");
+            return getAvailableMemoryWindowsPowerShell();
+        } catch (OperatingSystemException | MemoryInformationException e) {
+            // If PowerShell fails, fall back to Java MXBean
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "PowerShell failed, falling back to Java MXBean", e);
+            }
+			
+            // Note: JDK free memory does not include cached/standby memory, and might
+            // be lower than previous implementation (wmic) and new PowerShell results.
+            return getFreeMemoryJDK();
+        }
+    }
 
-            // https://technet.microsoft.com/en-ca/aa394268(v=vs.71)
-
-            for (String line : lines) {
-                if (line.startsWith("AvailableBytes=")) {
-                    available += processWmicLine(line);
-                } else if (line.startsWith("CacheBytes=")) {
-                    available += processWmicLine(line);
+    private long getAvailableMemoryWindowsPowerShell() throws OperatingSystemException, MemoryInformationException {
+        // Use PowerShell to get the same memory information that wmic provided
+        // Get-CimInstance Win32_PerfFormattedData_PerfOS_Memory provides AvailableBytes and CacheBytes
+        List<String> lines = OperatingSystem.executeProgram("powershell", "-NoProfile", "-Command",
+                "$mem = Get-CimInstance Win32_PerfFormattedData_PerfOS_Memory; " +
+                "Write-Output ($mem.AvailableBytes + $mem.CacheBytes)");
+        
+        for (String line : lines) {
+            line = line.trim();
+            if (!line.isEmpty() && line.matches("\\d+")) {
+                long available = Long.parseLong(line);
+                if (available > 0) {
+                    return available;
                 }
             }
-
-            if (available <= 0) {
-                throw new MemoryInformationException(Tr.formatMessage(tc, "memory.information.unexpected", "wmic", lines));
-            }
-
-            return available;
-        } catch (OperatingSystemException e) {
-            throw new MemoryInformationException(e);
         }
+        throw new MemoryInformationException(Tr.formatMessage(tc, "memory.information.unexpected", "PowerShell", lines));
     }
 
     @FFDCIgnore({ OperatingSystemException.class })

